pathio.go

```diff
@@ -114,6 +112,41 @@ func (c *Client) ListFiles(path string) ([]string, error) {
 	return lsLocal(path)
 }
 
+// Exists determines if a path does or does not exist.
+// NOTE: S3 is eventually consistent so keep in mind that there is a delay.
+func (c *Client) Exists(path string) (bool, error) {
+	if strings.HasPrefix(path, "s3://") {
+		s3Conn, err := s3ConnectionInformation(path, c.Region)
+		if err != nil {
+			return false, err
+		}
+		return existsS3(s3Conn)
+	}
+	return false, nil
+}
+
+func existsS3(s3Conn s3Connection) (bool, error) {
+	_, err := s3Conn.handler.HeadObject(&s3.HeadObjectInput{
+		Bucket: aws.String(s3Conn.bucket),
+		Key:    aws.String(s3Conn.key),
+	})
+	if err != nil {
+		if aerr, ok := err.(s3.RequestFailure); ok && aerr.StatusCode() == 404 {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+func existsLocal(path string) (bool, error) {
```

Comment:
I think I put the default response in `Exists` to get the linter to shut up will I was implementing `existsLocal`